include/aspect/particle/interpolator/interface.h

```diff
@@ -74,7 +75,34 @@ namespace aspect
           std::vector<std::vector<double> >
           properties_at_points(const std::multimap<types::LevelInd, Particle<dim> > &particles,
                                const std::vector<Point<dim> > &positions,
-                               const typename parallel::distributed::Triangulation<dim>::active_cell_iterator &cell = typename parallel::distributed::Triangulation<dim>::active_cell_iterator()) const = 0;
+                               const typename parallel::distributed::Triangulation<dim>::active_cell_iterator &cell = typename parallel::distributed::Triangulation<dim>::active_cell_iterator()) const;
+
+          /**
+           * Perform an interpolation of the properties of the particles in
+           * this cell onto a vector of positions in this cell.
+           * Implementations of this function must return a vector of a vector
+           * of doubles which contains a somehow computed
+           * value for all in @p selected_properties selected particle
+           * properties at all given positions in @p positions.
+           *
+           * @param [in] particles Reference to the particle map.
+           * @param [in] positions The vector of positions where the properties
+           * should be evaluated.
+           * @param [in] selected_properties A component mask that determines
+           * which particle properties are interpolated in this function.
+           * @param [in] cell An optional iterator to the cell containing the
+           * particles. Not all callers will know the cell of the particles,
+           * but providing the cell when known speeds up the interpolation
+           * significantly.
+           * @return A vector with as many entries as @p positions. Every entry
+           * is a vector of interpolated tracer properties at this position.
```

Comment:
And if the former, how do you map selected components onto the elements of the return vector?

I'm going to add that I typically prefer the case where the function always returns vectors of the same length, and simply only fills the elements of the vector that correspond to selected components. This avoids the question of how to map indices.